tests/bluechi_test/container.py

```diff
@@ -116,13 +116,26 @@ def get_unit_state(self, unit_name: str) -> str:
         _, output = self.exec_run(f"systemctl is-active {unit_name}")
         return output
 
-    def wait_for_unit_state_to_be(self, unit_name: str, expected_state: str, timeout: float = 4.0) -> bool:
+    def _is_unit_in_state(self, unit_name: str, expected_state: str) -> bool:
+        latest_state = self.get_unit_state(unit_name)
+        logger.debug(f"Got state '{latest_state}' for unit {unit_name}")
+        return latest_state == expected_state
+
+    def wait_for_unit_state_to_be(
+            self,
+            unit_name: str,
+            expected_state: str,
+            timeout: float = 5.0,
+            delay: float = 0.5) -> bool:
         latest_state = ""
 
+        if self._is_unit_in_state(unit_name, expected_state):
+            return True
+
         start = time.time()
         while (time.time() - start) < timeout:
-            latest_state = self.get_unit_state(unit_name)
-            if latest_state == expected_state:
+            time.sleep(delay)
```

Comment:
I think this functionality overlaps with what I did with `systemctl_start_and_wait` in #575 
Maybe `systemctl_start_and_wait` would be redundant after that

Reply:
When looking at our test code I was wondering if we could use `wait_for_unit_state_to_be` instead of hardcoded delay in `systemctl_start_and_wait`. I'm going to check it, but maybe we will need to implement `notify` service type functionality to our services: https://www.freedesktop.org/software/systemd/man/systemd.service.html#Options

Anyway at the moment this just improves existing code as we use `wait_for_unit_state_to_be`  much more often than `systemctl_start_and_wait`

Reply:
ack